integrationm/common/com/cultofbits/customizations/audit/AuditInstanceUpdater.groovy

```diff
@@ -0,0 +1,134 @@
+package com.cultofbits.customizations.audit
+
+import com.cultofbits.integrationm.service.actionpack.RecordmActionPack
+import com.google.common.cache.Cache
+import com.cultofbits.integrationm.service.dictionary.recordm.Definition
+import com.cultofbits.integrationm.service.dictionary.recordm.FieldDefinition
+import com.cultofbits.integrationm.service.dictionary.recordm.RecordmMsg
+import com.google.common.cache.CacheBuilder
+import org.json.JSONObject
+
+class AuditInstanceUpdater {
+
+
+    static Definition getDefinitionFromCache(cacheOfAuditFieldsForDefinition, RecordmMsg msg, RecordmActionPack recordm){
```

Comment:
You shouldn't pass the cache as an argument. This is bad design. This is exposing other state to other class when this method can invalidate state of the cache.

Reply:
I did that to decouple the cache from the method. What is your suggestion that I do ?